### DISCUSSION TITLE
bug: useContractEvent listener doesn't receive the full event

### DISCUSSION THREAD

**Discussion by DenisGorbachev** [wevm/wagmi#1799] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.11.4

### Current Behavior

The useContractEvent listener has the following signature:

```
listener: (...args: TArgs extends readonly unknown[] ? TArgs : unknown[]) => void;
```

It receives the args of the event, but not the full event object.

### Expected Behavior

We need to receive the following properties from the full event:

```
blockHash
blockNumber
transactionHash
transactionIndex
logIndex
```

They are otherwise unavailable, the only chance to receive them is in the event listener, but it doesn't pass the full event.

### Steps To Reproduce

Any useContractEvent call reproduces the issue.

**Comment by tmm** [comment]
Moving to feature request

**Reply by DenisGorbachev** [reply]
TBH it does feel like a bug. The full event is returned from a regular RPC call. The full event is not retruned from useContractEvent.

Anyway, thank you for a great library - it's so much better than the existing alternatives!

**Comment by alanrsoares** [comment]
Any update on this? I can't upgrade past v0.10.0. This breaking change is not documented anywhere. I was introduced on v0.10.1.

@DenisGorbachev did you find a workaround?

@tmm this shouldn't be tagged as `feature request` as it actually represents an undocumented breaking change

**Reply by DenisGorbachev** [reply]
Nope, I had to downgrade the `wagmi` package
